src/Tests/AuthenticationTrustedTests.cs

```diff
@@ -0,0 +1,206 @@
+using System;
+using Xunit;
+using FluentAssertions;
+
+using EmsApi.Client.V2;
+
+namespace EmsApi.Tests
+{
+    public class AuthenticationTrustedTests : TestBase
+    {
+        [Fact( DisplayName = "Client id but no client secret should fail" )]
+        public void Client_id_but_no_secret_should_fail()
+        {
+            EmsApiServiceConfiguration config = m_config.Clone();
+            config.UserName = null;
+            config.Password = null;
+            config.ApiClientId = "foobarbaz";
+            config.ApiClientSecret = null;
+            Action create = () => _ = new EmsApiService( config );
+            create.Should().Throw<EmsApiConfigurationException>();
+        }
+
+        [Fact( DisplayName = "Client secret but no id should fail" )]
+        public void Client_secret_but_no_id_should_fail()
+        {
+            EmsApiServiceConfiguration config = m_config.Clone();
+            config.UserName = null;
+            config.Password = null;
+            config.ApiClientId = null;
+            config.ApiClientSecret = "not so secret";
+            Action create = () => _ = new EmsApiService( config );
+            create.Should().Throw<EmsApiConfigurationException>();
+        }
+
+        [Fact( DisplayName = "Trusted value but no name should fail" )]
+        public void Trusted_value_but_no_name()
+        {
+            using var service = NewService();
+            var ctx = new CallContext { TrustedAuthValue = "ksk" };
+            Action getSystem = () => service.EmsSystem.Get( ctx );
+            getSystem.Should().Throw<EmsApiException>();
+        }
+
+        [Fact( DisplayName = "Trusted info but no client id should fail" )]
+        public void Trusted_info_but_no_client_id()
+        {
+            EmsApiServiceConfiguration config = m_config.Clone();
+            config.ApiClientId = null;
+            config.ApiClientSecret = null;
+            using var service = new EmsApiService( config );
+            var ctx = new CallContext
+            {
+                TrustedAuthName = "SAMAccountName",
+                TrustedAuthValue = "ksk"
```

Comment:
I have a service user in the US region w/ the name `emsapitest`, we should use that for the tests that will make a real call

Reply:
Good call. We need two users, Cody and I decide on `emsapitest` and `emsweb` as those should stick around.